imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -0,0 +1,115 @@
+"""Calculates Extended Raw Events for ULTRA L1b."""
+
+import numpy as np
+
+from imap_processing.ultra.l1b.l1b_extended_config import (
+    D_SLIT_FOIL,
+    N_ELEMENTS,
+    SLIT_Z,
+    TRIG_CONSTANT,
+    YF_ESTIMATE_LEFT,
+    YF_ESTIMATE_RIGHT,
+)
+from imap_processing.ultra.l1b.lookup_utils import (
+    get_image_params,
+    get_y_adjust,
+)
+
+# TODO: make lookup tables into config files.
+
+
+def get_front_x_position(
+    start_type: np.array, start_position_tdc: np.array
+) -> np.array:
+    """
+    Calculate the front xf position.
+
+    Converts Start Position Time to Digital Converter (TDC)
+    values into units of hundredths of a millimeter using a scale factor and offsets.
+    Further description is available on pages 30 of
+    IMAP-Ultra Flight Software Specification document (7523-9009_Rev_-.pdf).
+
+    Parameters
+    ----------
+    start_type : np.array
+        Start Type: 1=Left, 2=Right.
+    start_position_tdc : np.array
+        Start Position Time to Digital Converter (TDC).
+
+    Returns
+    -------
+    xf : np.array
+        X front position (hundredths of a millimeter).
+    """
+    # Left and right start types.
+    indices = np.where((start_type == 1) | (start_type == 2))
+
+    xftsc = get_image_params("XFTSC")
+    xft_lt_off = get_image_params("XFTLTOFF")
+    xft_rt_off = get_image_params("XFTRTOFF")
+    xft_off = np.where(start_type[indices] == 1, xft_lt_off, xft_rt_off)
+
+    # Calculate xf and convert to hundredths of a millimeter
+    xf = (xftsc * -start_position_tdc[indices] + xft_off) * 100
+
+    return xf
+
+
+def get_front_y_position(
+    start_type: np.array, yb: np.array
+) -> tuple[np.array, np.array]:
+    """
+    Compute the adjustments for the front y position and distance front to back.
+
+    This function utilizes lookup tables and trigonometry based on
+    the angle of the foil. Further description is available in the
+    IMAP-Ultra Flight Software Specification document pg 30.
+
+    Parameters
+    ----------
+    start_type : np.array
+        Start Type: 1=Left, 2=Right.
+    yb : np.array
+        Y back position in hundredths of a millimeter.
+
+    Returns
+    -------
+    d : np.array
+        Distance front to back in hundredths of a millimeter.
+    yf : np.array
+        Front y position in hundredths of a millimeter.
+    """
+    # Determine start types
+    index_left = np.where(start_type == 1)
+    index_right = np.where(start_type == 2)
```

Comment:
numpy documentation suggests using `np.nonzero()` for getting indices. 
```suggestion
    index_left = np.nonzero(start_type == 1)
    index_right = np.nonzero(start_type == 2)
```